src/ReplaySession.cc

```diff
@@ -1018,6 +1033,8 @@ Completion ReplaySession::try_one_trace_step(
     // cont_syscall_boundary handles the ticks constraint for those cases.
     return advance_to_ticks_target(t, constraints);
   }
+  if (current_step.action != TSTEP_ENTER_SYSCALL)
+    clear_syscall_bp();
```

Comment:
Why do we need to clear it here?

AFAIK we won't start a new step until after enter_syscall has returned COMPLETE, in which case we should already have cleared it.